### PR TITLE
[SPARK-39437][SQL][TEST] Normalize plan id separately in PlanStabilitySuite

### PR DESCRIPTION
**Pull request by cloud-fan**
### What changes were proposed in this pull request?

In `PlanStabilitySuite`, we normalize expression IDs by matching `#\d+` in the explain string. However, this regex can match plan id in `Exchange` node as well, which will mess up the normalization if expression IDs and plan IDs overlap.

This PR normalizes plan id separately in `PlanStabilitySuite`.
 
### Why are the changes needed?

Make the plan golden file more stable.

### Does this PR introduce _any_ user-facing change?

no

### How was this patch tested?

N/A

### CODE REVIEW

**Comment by cloud-fan** [comment]
cc @dongjoon-hyun

File: sql/core/src/main/scala/org/apache/spark/sql/execution/exchange/Exchange.scala [hunk 1]
```diff
@@ -37,7 +37,7 @@ abstract class Exchange extends UnaryExecNode {
   override def output: Seq[Attribute] = child.output
   final override val nodePatterns: Seq[TreePattern] = Seq(EXCHANGE)
 
-  override def stringArgs: Iterator[Any] = super.stringArgs ++ Iterator(s"[id=#$id]")
+  override def stringArgs: Iterator[Any] = super.stringArgs ++ Iterator(s"[plan_id=$id]")
```
Comment (dongjoon-hyun) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/exchange/Exchange.scala:40]: Thank you so much!

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Yes, I expected this kind of massive PR. :)

**Review by HyukjinKwon — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Could you fix more test failures?
```
24881 tests run, 682 skipped, 7 failed.
```

**Comment by cloud-fan** [comment]
More golden files need to be re-generated. Working on it now.

**Comment by dongjoon-hyun** [comment]
Oh, `q83` becomes flaky in `GitHub Action` environment on `master` branch.
- https://github.com/apache/spark/commit/a3988bcaa3ccc57c3bf34273ad48c6405673fc90
```
[info] - check simplified (tpcds-v1.4/q83) *** FAILED *** (182 milliseconds)
...
[info] - check simplified sf100 (tpcds-v1.4/q83) *** FAILED *** (183 milliseconds)
...
[info] *** 2 TESTS FAILED ***
[error] Failed: Total 1721, Failed 2, Errors 0, Passed 1719, Ignored 2
[error] Failed tests:
[error] 	org.apache.spark.sql.TPCDSV1_4_PlanStabilitySuite
[error] 	org.apache.spark.sql.TPCDSV1_4_PlanStabilityWithStatsSuite
```

I verified that it's stable locally.
```
$ SPARK_GENERATE_GOLDEN_FILES=1 build/sbt "sql/testOnly *PlanStabilitySuite"
$ git status
On branch master
Your branch is up to date with 'apache/master'.

nothing to commit, working tree clean

$ SPARK_GENERATE_GOLDEN_FILES=1 build/sbt "sql/testOnly *PlanStabilitySuite -- -z (tpcds-v1.4/q83)"
$ git status
On branch master
Your branch is up to date with 'apache/master'.

nothing to commit, working tree clean
```

**Comment by dongjoon-hyun** [comment]
Ah, I found that it is ANSI mode test. Let me make a follow-up.

**Pull request closed without merging**